File: sample/src/main/java/com/asual/summer/sample/domain/License.java
package com.asual.summer.sample.domain;

import static javax.persistence.GenerationType.IDENTITY;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EntityManager;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.PersistenceContext;
import javax.persistence.Table;

import org.hibernate.validator.constraints.NotEmpty;
import org.springframework.beans.factory.annotation.Configurable;
import org.springframework.transaction.annotation.Transactional;

import com.asual.summer.core.util.StringUtils;

@Configurable
@Entity
@Table
public class License implements java.io.Serializable {

    private static final long serialVersionUID = 1L;

    @PersistenceContext
    private transient EntityManager entityManager;
    
    @Id
    @GeneratedValue(strategy=IDENTITY)
    @Column(unique=true, nullable=false)
    private Integer id;

    @NotEmpty
    @Column(unique=true, nullable=false)    
    private String value;
    
    @NotEmpty
    @Column
    private String name;
    
    public License() {
    }
    
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        if (name != null) {
            value = StringUtils.toURIPath(name);
        }
        this.name = name;
    }
    
    public boolean equals(Object obj) {
        if (obj == null)
            return false;
        if (obj == this)
            return true;
        if (obj.getClass() != getClass())
            return false;
        return value != null && value.equals(((License) obj).value);
    }
    
    @Transactional
    public void persist() {
        entityManager.persist(this);
    }
    
    @Transactional
    public License merge() {
    	License merged = entityManager.merge(this);
        entityManager.flush();
        return merged;
    }
    
    @Transactional
    public void remove() {
        if (entityManager.contains(this)) {
            entityManager.remove(this);
        } else {
        	License attached = entityManager.find(this.getClass(), this.id);
            entityManager.remove(attached);
        }
    }
    
    @Transactional
    public void flush() {
        entityManager.flush();
    }

    public static final EntityManager entityManager() {
        EntityManager em = new License().entityManager;
        if (em == null) {
        	throw new IllegalStateException("Entity manager has not been injected.");
        }
        return em;
    }
    
    public static License find(String value) {
    	return (License) entityManager().createQuery("select o from License o where o.value = ?1").setParameter(1, value).getSingleResult();
    }
    
    @SuppressWarnings("unchecked")
    public static List<License> list() {
        return entityManager().createQuery("select o from License o").getResultList();
    }
    
    @SuppressWarnings("unchecked")
    public static List<License> list(int firstResult, int maxResults) {
        return entityManager().createQuery("select o from License o").setFirstResult(firstResult).setMaxResults(maxResults).getResultList();
    }

}

File: samples/scala/src/main/scala/com/asual/summer/sample/domain/License.scala
/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *	  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.asual.summer.sample.domain

import com.asual.summer.core.util.StringUtils

import java.lang.Integer
import java.util.List

import javax.persistence._

import org.hibernate.validator.constraints.NotEmpty
import org.springframework.beans.factory.annotation.Configurable
import org.springframework.transaction.annotation.Transactional

import scala.reflect.BeanProperty

/**
 * 
 * @author Rostislav Hristov
 *
 */
@Configurable
@Entity
@Table
@SerialVersionUID(1L)
@serializable
class License {

	@PersistenceContext
	@BeanProperty
	@transient 
	var entityManager:EntityManager = _
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(unique=true, nullable=false)
	@BeanProperty
	var id:Integer = _

	@NotEmpty
	@Column(unique=true, nullable=false)
	@BeanProperty
	var value:String = _
	
	@NotEmpty
	@Column
	@BeanProperty
	var name:String = _

	override def hashCode = {
		41 * value.hashCode
	}
	
	override def equals(other:Any) = other match {
		case that: License => 
			(that canEqual this) && (this.value == that.value)
		case _ => 
			false
	} 
	
	def canEqual(other:Any) = {
		other.isInstanceOf[License]
	}

	@Transactional
	def persist = entityManager.persist(this)
	
	@Transactional
	def merge:License = {
		var merged:License = entityManager.merge(this)
		entityManager.flush
		return merged
	}
	
	@Transactional
	def remove = {
		if (entityManager.contains(this)) {
			entityManager.remove(this)
		} else {
			var attached:License = 
				entityManager.find(this.getClass(), this.id).asInstanceOf[License]
			entityManager.remove(attached)
		}
	}
	
	@Transactional
	def flush = {
		entityManager.flush
	}	
}

object License {
	
	def entityManager:EntityManager = {
		var em:EntityManager = new License().entityManager
		if (em == null) {
			throw new IllegalStateException("Entity manager has not been injected.")
		}
		return em
	}
	
	def find(value:String):License = {
		return entityManager.createQuery("select o from License o where o.value = ?1")
			.setParameter(1, value).getSingleResult.asInstanceOf[License]
	}
	
	def list():List[License] = {
		return entityManager.createQuery("select o from License o")
			.getResultList.asInstanceOf[List[License]]
	}
	
	def list(firstResult:Int, maxResults:Int):List[License] = {
		return entityManager.createQuery("select o from License o")
			.setFirstResult(firstResult).setMaxResults(maxResults)
				.getResultList.asInstanceOf[List[License]]
	}

}